rust_src/remacs-sys/lib.rs

```diff
@@ -931,6 +942,52 @@ extern "C" {
     pub fn Ffuncall(nargs: ptrdiff_t, args: *mut Lisp_Object) -> Lisp_Object;
     pub fn Fpurecopy(string: Lisp_Object) -> Lisp_Object;
 
+    pub fn Fexpand_file_name(name: Lisp_Object, default_directory: Lisp_Object) -> Lisp_Object;
+    pub fn Ffind_file_name_handler(filename: Lisp_Object, operation: Lisp_Object) -> Lisp_Object;
+
+    pub fn re_search(
+        bufp: *mut re_pattern_buffer,
+        string: *const c_char,
+        size: size_t,
+        startpos: ssize_t,
+        range: ssize_t,
+        regs: *mut re_registers,
+    ) -> ssize_t;
+
+    pub fn compile_pattern(
+        pattern: Lisp_Object,
+        regp: *mut re_registers,
+        transmute: Lisp_Object,
+        posix: bool,
+        multibyte: bool,
+    ) -> *mut re_pattern_buffer;
+
+    pub fn filemode_string(filename: Lisp_Object) -> Lisp_Object;
+    pub fn call2(func: Lisp_Object, arg1: Lisp_Object, arg2: Lisp_Object) -> Lisp_Object;
```

Comment:
We have the `call!` macro to handle not having to use these functions directly, I think that we should prefer to use that over direct calls to call2, call3 etc.

Reply:
Good catch, totally missed it. I'll try it and resubmit